### ISSUE TITLE
 Unusual routing decision

### ISSUE THREAD

**Issue by scaidermern** [Project-OSRM/osrm-backend#4754]
Issue forwarded from https://help.openstreetmap.org/questions/61189/unusual-routing-decision

Expected behavior:
https://www.openstreetmap.org/directions?engine=osrm_car&route=50.85216%2C-1.16498%3B50.85311%2C-1.16852#map=18/50.85265/-1.16674
![foo01](https://user-images.githubusercontent.com/1040784/34294919-03264fa6-e70b-11e7-870b-fc481b712eff.png)

Unexpected behavior:
https://www.openstreetmap.org/directions?engine=osrm_car&route=50.85217%2C-1.16484%3B50.85311%2C-1.16852#map=18/50.85265/-1.16674
![foo02](https://user-images.githubusercontent.com/1040784/34294930-0925dac0-e70b-11e7-90c5-8723c65a84b0.png)

The starting position in the second case is only slightly altered and includes an additional traffic signal. It leads however to a completely different route which is 6.9km long (time: 0:11) instead of only 276m (time: 0:03).

Other routing engines (GraphHopper, Mapzen) don't show this behavior.

The map data seems to be fine. Routing across the traffic signal is not an option and works perfectly:
https://www.openstreetmap.org/directions?engine=osrm_car&route=50.85217%2C-1.16484%3B50.85232%2C-1.16560#map=18/50.85265/-1.16674
![foo03](https://user-images.githubusercontent.com/1040784/34295080-b2d189a2-e70b-11e7-9c80-43181ff41a93.png)

**Comment by daniel-j-h** [comment]
Here's the router's response for the first request:

    curl 'https://router.project-osrm.org/route/v1/driving/-1.16498,50.85216;-1.16852,50.85311' | jq '.routes[0] | {weight: .weight, duration: .duration, distance: .distance}'



```json
{
  "weight": 912.2,
  "duration": 178.5,
  "distance": 275.5
}
```

And here is the response for the second query

    curl 'https://router.project-osrm.org/route/v1/driving/-1.16484,50.85217;-1.16852,50.85311' | jq '.routes[0] | {weight: .weight, duration: .duration, distance: .distance}'

```json
{
  "weight": 886.3,
  "duration": 638.2,
  "distance": 6862.3
}
```

For some reason the route's weight we optimize for is smaller for the second route, even though duration and distance are higher. I quickly skimmed the data but couldn't find anything suspicious.

Traffic signals change duration here:

https://github.com/Project-OSRM/osrm-backend/blob/84b6ef43403b75ec683a4e50dde58b20c476ac24/src/extractor/edge_based_graph_factory.cpp#L578-L597

https://github.com/Project-OSRM/osrm-backend/blob/84b6ef43403b75ec683a4e50dde58b20c476ac24/profiles/car.lua#L426-L428

**Comment by scaidermern** [comment]
Shouldn't lead a traffic light penalty to a higher duration? The duration of the second query is higher than the first one. So this can't be caused by traffic light penalties, can it?

**Comment by TheMarex** [comment]
I found the origin of this problem that is the result of an invalidly applied U-Turn penalty. This is amplified by the fact that the demo sever uses 60 seconds for the U-Turn penalty, as opposed to 20 seconds like the stock profile. A fix for this is in #4755 and will be shipped with the next bug fix release.

**Comment by oxidase** [comment]
Fixed in v5.14.3
